geomas/core/vision/encoder/api_encoder.py

```diff
@@ -0,0 +1,61 @@
+from typing import Dict, List
+import torch
+import requests
+import time
+from tqdm import tqdm
+from math import ceil
+import numpy
+from geomas.core.vision.encoder.encoder import Encoder
+
+
+class ApiEncoder(Encoder):
+    @staticmethod
+    def code_name() -> str:
+        return "api"
+
+    def __init__(self, model_name, **kwargs):
+        super().__init__()
+        self.model_name = model_name
+        self.headers = kwargs.get("headers", {"Content-Type": "application/json"})
+        self.max_retries = kwargs.get("max_retries", 3)
+        self.retry_delay = kwargs.get("retry_delay", 5)
+        self.max_length = kwargs.get("max_length", 1024)
+
+    def get_embedding(self, texts: list):
+        attempt = 0
+        while attempt < self.max_retries:
+            try:
+                response = requests.post(
+                    self.model_name,
+                    headers=self.headers,
+                    json={"input": texts},
+                    verify=False,
+                )
+                response.raise_for_status()
+                embeddings = [
+                    data["embedding"] for data in response.json().get("data", [])
+                ]
+                return numpy.array(embeddings)
+
+            except requests.exceptions.RequestException as e:
+                attempt += 1
+                print(
+                    f"Error during API request (Attempt {attempt}/{self.max_retries}): {e}"
+                )
```

Comment:
here specifically and in all such cases, try not to output anything to standard output via `print`
would be better to replace this with the use of logging with a selected logging level

```suggestion
                logger.error(f"Error during API request (Attempt {attempt}/{self.max_retries}): {e}")
```